docs/meetings/exeter-jun-2019.md

```diff
@@ -0,0 +1,254 @@
+
+- UI: prioritize Tree View based on OS mock-ups. Dot-view is low priority.
+
+- UIS-WFS authentication/trust - similar to CLI, e.g. one-time or timed token.
+
+- need encryption between Hub and UIS (not the default in J-Hub).
+  - high priority once we have the end-to-end system
+
+- UIS to do all authorization if possible. It has to do some, e.g. for suite
+  start-up, so might as well do it all.
+  - so audit trail for all client interaction needed at the UIS
+
+- User feedback request on Discourse
+  - agreed we should do this
+  - not wide spectrum of use cases - can't promise to please everyone
+  - SB to write up for review before posting to Discourse.
+
+- We will rely on J-Hub to be present, even for single-user installations
+
+- UIS needs to abort (or prob just be useless) if not started/owned by the Hub.
+
+- we may eventually need a native e-app (e.g. electron) for those without the
+  right network routing.
+
+- JupyterLab is a different/better notebook interface. Has a nice user-movable
+  tiling tab feature that we should try to use. 
+
+- Most tree view info should be retrieved on request only (e.g. Job ID etc. in
+  the current tree view, not needed unless the final level is expanded in the
+  new UI.
+
+- "rose suite-run" - need plugin-based file-installation, e.g. via fcm keywords
+
+- rose-suite.conf should be an importable Python file.
+
+- UIS has to see source dir
+  - need to see diffs of src vs run-dir
+  - MS: could use git (whole suite, minus work tree etc.); tie commits to runs,
+    restarts, reloads.
+
+- Cylc 9:
+  Python API gives modularity
+  parameterizing sub-graphs 
+  spawn-on-demand: avoids need for multi-dimensional cycling (just follow the
+   graph, whether it is generated by cycles or parameters)
+
+- Agreed new default installation dir structure:
+```
+    src ---> src/run1
+                /run2
+                /run3
+```
+  -  every cold-start creates a new run-dir by default
+  - with symlink to src, so that cylc knows it is a template suite.
+  - reload should show a diff of old vs new
+  - in the UI gscan sidebar: collapse on src suite; select src for a new
+    run; select run1 to stop suite etc.
+  - housekeeping: select to delete runX
+  - show latest runs and/or running suites by default
+  - show out-of-date config (with respect to src)
+  - only support editing the source suite via UI, at least initially
+    - although separate run-dirs make run dir editing safer now
+    - edit run-dir, save and reload
+  - need to be able to validate and graph the src dir (unlike now)
+
+- UI form for Jinja2 params
+- Note need static graph vis too, in Cylc 8 (tell MR)
+
+- TODO: can we "query the API" over GraphQL (UIS-WFS) so to allow multiple
+  cylc-flow versions on the back end?
+
+- OS agrees with DS PRs in large part; GraphQL schema looks good.
+  - not convinced on protobuf yet (but see below)
+
+- "stop suite" should report to user "waiting for active tasks to finish" (e.g.)
```

Comment:
And:
* report the list of active tasks it is waiting for?
* provide hint to users about other stop options?

Reply:
(See also #1032. Reason for shutdown should be recorded.)

Reply:
Noted.